storybook: initial setup w/ token spec generation

depatchedmode opened the pull request:
Fixes #738 

So far this just adds Storybook with the Design Token add-on and quickly instruments the `variables.scss` file for an example. 

Next on the list:

- [x] add an example story for the Button component
- [x] await feedback on new [shade palette and type scale proposal](https://www.figma.com/file/Q2UWiIm6FOOBmscXrMVid6/WG-9.4?node-id=657%3A2). This has now been accepted into the post-hicathon [UI/UX WG designs](https://www.figma.com/file/R5sw1HcoHx4Aup1Yl7rGYA/WG-UI-UX?node-id=1261%3A11247): 
- [ ] add these new tokens to the spec
~~- [ ] audit and document any other css custom properties / scss variables that should be considered core design tokens~~
~~- [ ] breakpoints audit~~
~~- [ ] basic layout grid~~

depatchedmode committed:
storybook: initial setup w/ token spec generation

andrevenancio commented:
So I'm all in to bring Storybook into the project. But..
The design done during hickaton is done by the community. Pretty good stuff in there, however, for the last couple of week we have a dedicated team of designers that have been working on a new version of the design.

I'm not the one who should be deciding what design needs to be implemented. Its my understanding that Raphael (hickaton) has been in touch with Julio (Designer) but I'm not sure how they're going to move forward.

I would not make major changes in new components until that has been agreed.

@crzypatchwork would you mind giving Julio and Raphael some direction of what would you like to keep from the 2 different designes we have at the moment?

andrevenancio commented on package.json:
```diff
@@ -49,10 +49,22 @@
     "eb:prod": "npm run build && npm run start:prod",
-    "prettier": "prettier --write \"./src/**/*.{js,jsx}\""
+    "prettier": "prettier --write \"./src/**/*.{js,jsx}\"",
+    "storybook": "start-storybook -p 6006 -s public",
```
Can you remove `public` ? we dont want it to build to the public folder but rather for Storybook folder `storybook-static`. This will allow us to coordinate 2 different deployments. The public folder should only be for the react-create-app application

andrevenancio commented on src/stories/tokens-animation.stories.mdx:
```diff
@@ -0,0 +1,6 @@
+import { Meta } from '@storybook/addon-docs/blocks';
+import { DesignTokenDocBlock } from 'storybook-design-token/dist/doc-blocks';
+
+<Meta title="Design Tokens/Animation" />
```
Would be nice to also use the `controls` feature from Storybook

depatchedmode replied:
I believe those are included by default and bind automatically where supported now:
https://storybook.js.org/docs/react/essentials/controls

depatchedmode replied:
Can still be configured custom, but probably a pattern to test when we get to the components phase.

andrevenancio replied:
What do you think of this pattern?
```
import { Story, Meta } from '@storybook/react/types-6-0';

import { Example, ExampleProps } from './index';

export default {
  title: 'Example',
  component: Example,
  args: {
    children: 'An example',
  },
} as Meta;

const Template: Story<ExampleProps> = (args: any) => {
  return <Example {...args} />;
};

export const Basic = Template.bind({});

```

depatchedmode replied:
No complaints here.

depatchedmode replied:
ok - wasn't seeing much movement yet on the Figma design system proposals, so thought it might just be good to add an example story for the Button component.

I don't have a ton of recent React experience, but I followed the pattern above as best I could minus the Typescript aspects which I don't think are established in this repo yet.

Let me know if you want anything tweaked.

depatchedmode commented:
@andrevenancio Fully agree. Wasn't intending for this PR to deal w components at all, yet. Just to formalize design tokens, and that first to do is just noting that there are some proposals that need to be agreed upon, pending feedback from the core team. 

I saw that you raised concerns in the Figma about contrast and I agree. That's why I put together that "shade palette" proposal with WCAG AA/AAA contrast ratios details for both the light and dark mode themes. If the core team has a different one, also cool.

Just want to ensure that fundamental palette of shades and type sizes is nailed for everyone in branch. Feels like that bit is achievable and is the best first Story.

depatchedmode commented:
@andrevenancio oh one other question I did have:

there are both scss variables and css custom properties in the codebase. it looks like custom properties is the preferred approach, as that's used in the components. are you moving away from scss or is the intention to use both?

andrevenancio commented:
> @andrevenancio oh one other question I did have:
> 
> there are both scss variables and css custom properties in the codebase. it looks like custom properties is the preferred approach, as that's used in the components. are you moving away from scss or is the intention to use both?

So the project started with css. I moved in and implemented sass, there might be some old css laying around.

We've been talking about moving this project to next.js. So that's something to consider too.

depatchedmode committed:
storybook: change build target to storybook-static

andrevenancio commented on package.json:
```diff
@@ -49,10 +49,22 @@
     "eb:prod": "npm run build && npm run start:prod",
-    "prettier": "prettier --write \"./src/**/*.{js,jsx}\""
+    "prettier": "prettier --write \"./src/**/*.{js,jsx}\"",
+    "storybook": "start-storybook -p 6006 -s storybook-static",
```
this is fine too, but if you just add  `start-storybook -p 6006` it will default to `storybook-static`.

andrevenancio reviewed (approved):
all looks fine. just some minor comments.

I would probably focus on the src/components/layout

depatchedmode committed:
add an example Button story

depatchedmode committed:
clean up storybook script

crzypatchwork commented:
this was an interesting thing for documenting the protocol if I understood it right

depatchedmode commented:
> this was an interesting thing for documenting the protocol if I understood it right

Yeah, the MDX files would be great for that in addition to documenting the design system.

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

joao committed:
Missing space in About text

After the reference to Telegram there should be a non-breaking space inserted, as to separate the two words: 'telegram' and 'or'.

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

crzypatchwork committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

etienne

kylegrover committed:
Update w.json

polygon

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

crzypatchwork committed:
Update w.json

kylegrover committed:
Merge pull request #1099 from joao/patch-1

Missing space in About text

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

kylegrover committed:
fix-menu-top-spacing

kylegrover committed:
Update w.json

Commit:
music fix

Commit:
music fix

Commit:
hdao search fix

kylegrover committed:
Update w.json

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

somaticbits committed:
Update w.json

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

rkoshino committed:
Update w.json

rkoshino committed:
Update o.json

Commit:
1d1w

kylegrover committed:
subjkt validation

kylegrover committed:
Update w.json

gemz

crzypatchwork committed:
Merge pull request #1101 from hicetnunc2000/fix-menu-top-spacing

fix-menu-top-spacing

crzypatchwork committed:
Merge pull request #1103 from kylegrover/config-subjkt-validation-again

subjkt validation

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

somaticbits committed:
Update w.json

crzypatchwork committed:
Update w.json

kylegrover committed:
Update w.json

ibrajeez

kylegrover committed:
Update w.json

ibrajeez

kylegrover committed:
Update w.json

not jjjjjohn

krhoda committed:
adds discord support from TZProfiles

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

https://twitter.com/burkabayramart/status/1430286336410456066

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

dependabot[bot] committed:
Bump ipfs-http-client from 50.1.2 to 52.0.2

Bumps [ipfs-http-client](https://github.com/ipfs/js-ipfs) from 50.1.2 to 52.0.2.
- [Release notes](https://github.com/ipfs/js-ipfs/releases)
- [Changelog](https://github.com/ipfs/js-ipfs/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ipfs/js-ipfs/compare/ipfs-http-client@50.1.2...ipfs-http-client@52.0.2)

---
updated-dependencies:
- dependency-name: ipfs-http-client
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

kylegrover committed:
Update w.json

steamboy copy

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

https://twitter.com/Browtweaten/status/1430236952264155136

kylegrover committed:
Update w.json

history

kylegrover committed:
Update w.json

history

kylegrover committed:
Update w.json

miguelzinhe commented:
> > this was an interesting thing for documenting the protocol if I understood it right
> 
> Yeah, the MDX files would be great for that in addition to documenting the design system.

I can help with this addition!

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

https://www.pexels.com/@alice-shardan-72935338

kylegrover committed:
Update w.json

https://www.pexels.com/photo/woman-in-white-dress-shirt-3146257/

kylegrover committed:
Update w.json

https://www.pexels.com/search/extraterrestrial/

kylegrover committed:
Update w.json

violetforest committed:
add banner on homepage and profile page

crzypatchwork committed:
Merge pull request #1115 from violetforest/feature/banner-update

add banner on homepage and profile page

Commit:
logo

Commit:
remove banner

kylegrover committed:
Update w.json

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

kylegrover committed:
REACT_APP_GRAPHQL_API env var

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

temp unban melted punks

kylegrover committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update o.json

crzypatchwork committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

Commit:
small optimization?

Commit:
lowercase

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

runescape assets lol

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Update w.json

kylegrover committed:
Update w.json

stu ballinger

kylegrover committed:
Update w.json

john

kylegrover committed:
Update w.json

dualvoid

kylegrover committed:
Update w.json

unban, it's not copymint it's an old/hacked account

kylegrover committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

crzypatchwork committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

crzypatchwork committed:
Merge pull request #1111 from hicetnunc2000/dependabot/npm_and_yarn/ipfs-http-client-52.0.2

Bump ipfs-http-client from 50.1.2 to 52.0.2

crzypatchwork committed:
Merge pull request #1116 from kylegrover/add-graphqlapi-env-var

REACT_APP_GRAPHQL_API env var

crzypatchwork committed:
Merge pull request #1105 from spruceid/main

Feature - Add Discord Verification From Tezos Profiles to HEN

crzypatchwork committed:
Merge pull request #1084 from gschian0/main

Added P5 SVG Template

crzypatchwork committed:
Merge pull request #1087 from hicetnunc2000/dependabot/npm_and_yarn/taquito/taquito-10.1.0

Bump @taquito/taquito from 10.0.0 to 10.1.0

crzypatchwork committed:
Merge pull request #1086 from hicetnunc2000/dependabot/npm_and_yarn/taquito/beacon-wallet-10.1.0

Bump @taquito/beacon-wallet from 10.0.0 to 10.1.0

crzypatchwork committed:
Update w.json

kylegrover committed:
Update w.json

dualvoid

kylegrover committed:
Update w.json

mattbracini

kylegrover committed:
Update w.json

various stolen works

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

unban windowpunks

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

iamlirona

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

crzypatchwork committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

obstropolos committed:
added basic GitHub integration from TZP

kylegrover committed:
Update w.json

toshio

kylegrover committed:
Update w.json

stephenong

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

jjjjjjjjjjohn

kylegrover committed:
Update w.json

polygon

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
no number scrolling on swaps

kylegrover committed:
Update w.json

john

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

obstropolos committed:
Merge pull request #1 from spruceid/feat/github-verification

Add GitHub Verification to HEN from TZP

Commit:
git ignore

Commit:
git ignore

Commit:
git ignore

Commit:
.env.dev

kylegrover committed:
Update w.json

Commit:
nft.storage

crzypatchwork committed:
Update w.json

Commit:
nft.storage

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

kylegrover committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

kylegrover committed:
Update w.json

crzypatchwork committed:
Merge pull request #1124 from spruceid/main

Feature - Add GitHub Verification From Tezos Profiles to HEN

crzypatchwork committed:
Merge pull request #1121 from kylegrover/f-prevent-number-scrolling

prevent number scrolling

somaticbits committed:
Update w.json

crzypatchwork committed:
Update w.json

Commit:
env

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

Commit:
pdf fix

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

IntegroLabs committed:
switch index to hdapi.teztools.io, add hdapi.teztools.io to connect-src list

crzypatchwork committed:
Update w.json

crzypatchwork committed:
Merge pull request #1134 from IntegroLabs/main

switch index to hdapi.teztools.io, add hdapi.teztools.io to connect-s…

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

somaticbits committed:
Update w.json

crzypatchwork committed:
Update w.json

depatchedmode commented:
I've rebased onto `main`, and resolved the conflicts. Added a bit back to the scope of this, as it seems like it's worth getting the design tokens into this as a part of the post-hicathon UI/UX working group's efforts.

If these simple design tokens are accepted into main, it will make the other PR's (nav, minting page, etc) easier to implement and more consistent with each other.

Design tokens will be:
+ shade palette (light & dark mode)
+ type sizes
+ spacing units
